schema.json

```diff
@@ -398,6 +398,42 @@
       "required": ["symbol"],
       "type": "object"
     },
+    "string_argument": {
+      "additionalProperties": false,
+      "description": "A step argument containing a string.",
+      "properties": {
+        "string": {
+          "description": "The value. The string must be valid.",
+          "type": "string"
+        }
+      },
+      "required": ["string"],
+      "type": "object"
+    },
+    "number_argument": {
+      "additionalProperties": false,
+      "description": "A step argument containing a number.",
+      "properties": {
+        "number": {
+          "description": "The value. The number must be valid.",
+          "type": "number"
+        }
+      },
+      "required": ["number"],
```

Comment:
Historically, determining the parsing for the various numeric types (specifically U64) has been tricky due to lack of support by tooling (JS/TS can't handle U64 as a numeric and needs to use BigInt which has bad interop). Might want to consider a future where we specify the numeric types more specifically here.

Reply:
Yes 100% agree. I think this change sets us up nicely to provide a variety of more types in the future.

Reply:
Not sure if this is related but I remember MSLICE had the same issue with number precision between the client and seqgen and the satf. I remember multiple ISA from this. For example. 0.0000000018545 was truncated to  0.000000001 which some can argue was such a tiny number it shouldn't matter but ISA was created regardless.